java/statestore/src/main/java/sleeper/statestore/s3/S3PartitionStore.java

```diff
@@ -254,17 +252,6 @@ public void initialise(List<Partition> partitions) throws StateStoreException {
     }
 
     private void setPartitions(List<Partition> partitions) throws StateStoreException {
-        // Validate that there is no current revision id
-        Map<String, AttributeValue> item = new HashMap<>();
-        item.put(REVISION_ID_KEY, new AttributeValue().withS(CURRENT_PARTITIONS_REVISION_ID_KEY));
-        GetItemRequest getItemRequest = new GetItemRequest()
-                .withTableName(dynamoRevisionIdTable)
-                .withKey(item);
-        GetItemResult getItemResult = dynamoDB.getItem(getItemRequest);
-        if (null != getItemResult.getItem()) {
-            throw new StateStoreException("Dynamo should not contain current revision id; found " + item);
-        }
-
```

Comment:
Do we definitely want to remove the above? It means that if anyone uses the S3PartitionStore directly calls initialise() there is no check that it doesn't already contain partitions. The initialise method on the DynamoDB state store deletes any partitions when the initialise method is called, which is inconsistent with both the existing approach here and the above changes.

Reply:
It might be worth reworking the state stores to avoid you being able to call the S3PartitionStore directly.

This code does effectively delete the old partitions, as it generates the full state file for the partitions and sets that as the current revision. This is the same way any data is deleted in the S3 state store. It doesn't seem like we have any way to clean up old revisions right now. Should we look at adding that?

Reply:
I've made the FileInfoStore and PartitionStore implementations private to avoid this problem.